packages/react-tooltip/src/components/Tooltip/Tooltip.types.ts

```diff
@@ -23,67 +17,87 @@ export type TooltipProps = ComponentPropsCompat &
       | ((props: TooltipTriggerProps) => React.ReactNode)
       | null;
+  };
+};
 
-    /**
-     * The content displayed inside the tooltip.
-     */
-    content: ShorthandPropsCompat<ComponentPropsCompat>;
+/**
+ * Properties and state for Tooltip
+ */
+export interface TooltipCommons {
+  /**
+   * A tooltip can appear with the default appearance or inverted.
+   * When not specified, the default appearance is used.
+   */
+  appearance?: 'inverted';
+  /**
+   * The content displayed inside the tooltip.
+   */
+  content: React.ReactNode;
 
-    /**
-     * Render an arrow pointing to the target element
-     *
-     * @defaultvalue false
-     */
-    withArrow?: boolean;
+  /**
+   * Color variant with inverted colors
+   *
+   * @defaultvalue false
+   */
+  inverted?: boolean;
 
-    /**
-     * Control the tooltip's visibility programatically.
-     *
-     * This can be used in conjunction with onVisibleChange to modify the tooltip's show and hide behavior.
-     *
-     * If not provided, the visibility will be controlled by the tooltip itself, based on hover and focus events on the
-     * trigger (child) element.
-     */
-    visible?: boolean;
+  /**
+   * Render an arrow pointing to the target element
+   *
+   * @defaultvalue false
+   */
+  withArrow?: boolean;
 
-    /**
-     * Notification when the visibility of the tooltip is changing
-     */
-    onVisibleChange?: (
-      event: React.PointerEvent<HTMLElement> | React.FocusEvent<HTMLElement> | undefined,
-      data: OnVisibleChangeData,
-    ) => void;
+  /**
+   * Configure the positioning of the tooltip
+   *
+   * @defaultvalue above
+   */
+  positioning?: PositioningShorthand;
 
-    /**
-     * Specifies which aria attribute to set on the trigger element.
-     * * `label` - Set aria-label to the tooltip's content. Requires content to be a string; if not, uses `labelledby`.
-     * * `labelledby` - Set aria-labelledby to the tooltip's id. The id is generated if not provided.
-     * * `describedby` - Set aria-describedby to the tooltip's id. The id is generated if not provided.
-     * * null - Do not set any aria attributes on the trigger element.
-     *
-     * @defaultvalue label
-     */
-    triggerAriaAttribute?: 'label' | 'labelledby' | 'describedby' | null;
+  /**
+   * Control the tooltip's visibility programatically.
+   *
+   * This can be used in conjunction with onVisibleChange to modify the tooltip's show and hide behavior.
+   *
+   * If not provided, the visibility will be controlled by the tooltip itself, based on hover and focus events on the
+   * trigger (child) element.
+   */
+  visible?: boolean;
 
-    /**
-     * Delay before the tooltip is shown, in milliseconds.
-     *
-     * @defaultvalue 250
-     */
-    showDelay?: number;
+  /**
+   * Notification when the visibility of the tooltip is changing
+   */
+  onVisibleChange?: (
+    event: React.PointerEvent<HTMLElement> | React.FocusEvent<HTMLElement> | null,
```

Comment:
```suggestion
    event: React.PointerEvent<HTMLElement> | React.FocusEvent<HTMLElement> | undefined,
```

Let's apply `undefined` there, this fixes #19723.